src/tscore/runroot.cc

```diff
@@ -53,25 +52,38 @@ static std::string runroot_file = {};
 // the function for the checking of the yaml file in the passed in path
 // if found return the path to the yaml file, if not return empty string.
 static std::string
-get_yaml_path(const std::string &path, bool json)
+get_yaml_path(const std::string &path)
 {
+  // yaml_file 2 and 3 are for temporary use in case the change may break for people using runroot already
+  // this can be removed in the future.
   std::string yaml_file;
+  std::string yaml_file2;
+  std::string yaml_file3;
   if (ink_file_is_directory(path.c_str())) {
-    yaml_file = Layout::relative_to(path, "runroot_path.yml");
+    yaml_file  = Layout::relative_to(path, "runroot.yaml");
+    yaml_file2 = Layout::relative_to(path, "runroot.yml");
+    yaml_file3 = Layout::relative_to(path, "runroot_path.yml");
   } else {
-    if (path.substr(path.find_last_of("/") + 1) == "runroot_path.yml") {
+    std::string tmp = path.substr(path.find_last_of("/") + 1);
+    if (tmp == "runroot.yaml" || tmp == "runroot.yml" || "runroot_path.yml") {
       yaml_file = path;
     }
   }
   std::ifstream check_file;
+  std::ifstream check_file2;
+  std::ifstream check_file3;
   check_file.open(yaml_file);
-  if (!check_file.good()) {
-    if (!json) {
-      ink_warning("Unable to access runroot: '%s' - %s", yaml_file.c_str(), strerror(errno));
-    }
-    return {};
+  check_file2.open(yaml_file2);
```

Comment:
We should be using the filesystem wrapper we have. ie fs::exists(name) logic .. not trying to open it. We can give better error messages when it exists but has bad permission then just doing an open